### PR TITLE
[PLAT-369] Add noUncheckedIndexedAccess compiler flag

### PR DESCRIPTION
**Pull request by avatarneil**
## 📚 Purpose
This PR introduces the [noUncheckedIndexedAccess](https://www.typescriptlang.org/tsconfig#noUncheckedIndexedAccess) flag to the Typescript compiler in mds-core. This prevents unsafely accessing properties from arrays, and objects. It would have prevented bugs such as https://github.com/lacuna-tech/mds-core/pull/876. 

## 👌 Resolves:
- [x] PLAT-369
- [x] Lots of minor bugs that our previously unsafe accessing was covering up!

## 📦 Impacts:
Probably everything

### CODE REVIEW

**Review by nessur — APPROVED** [review]
this _is_ a lot, but awesome work. Thanks!

File: packages/mds-transaction-service/tests/transactions.spec.ts [hunk 1]
```diff
@@ -123,6 +134,9 @@ describe('Transaction Service Tests', () => {
 
         it('Create one transaction with missing fee_type rejects', async () => {
           const [transaction] = transactionsGenerator(1)
+          if (!transaction) {
+            throw new Error('No sample transaction found') // unexpected to ever happen
+          }
```
Comment (drtyh2o) [inline packages/mds-transaction-service/tests/transactions.spec.ts:139]: Seems like an opportunity for a function that asserts something is defined or throws that could replace all of the `if (!something) { throw new Error() }` blocks

**Review by drtyh2o — APPROVED** [review]
Phew! Thanks for all the work on this.

File: packages/mds-transaction-service/tests/transactions.spec.ts (hunk 1 above)
Comment (avatarneil) [reply packages/mds-transaction-service/tests/transactions.spec.ts:139]: Mmm yes good call, I'll file a followup ticket

**Pull request merged**